src/EnergyPlus/UnitarySystem.cc

```diff
@@ -17856,5 +17891,147 @@ namespace UnitarySystems {
+    {
+        auto outdoorAirController = state.dataMixedAir->OAController(this->OAControllerIndex);
+        using Psychrometrics::PsyCpAirFnW;
+        Real64 cpAir = PsyCpAirFnW(state.dataLoopNodes->Node(outdoorAirController.InletNode).HumRat);
+        Real64 zoneTemp = state.dataLoopNodes->Node(state.dataZoneEquip->ZoneEquipConfig(this->ControlZoneNum).ZoneNode).Temp;
+        // determine and set new air loop mixed air flow rate
+        Real64 mixedAirFlowRate = this->m_EconoPartLoadRatio * this->m_CoolMassFlowRate[this->m_EconoSpeedNum] +
+                                  this->m_LowSpeedEconRuntime * this->m_CoolMassFlowRate[this->m_EconoSpeedNum - 1];
+        if (airLoopNum > 0) {
+            // request fixed mixed flow rate
+            state.dataAirLoop->AirLoopControlInfo(airLoopNum).LoopFlowRateSet = true;
+            state.dataAirLoop->AirLoopFlow(airLoopNum).ReqSupplyFrac = mixedAirFlowRate / this->m_CoolMassFlowRate[this->m_NumOfSpeedCooling];
+            // adjust mixed air flow rate for rated air flow rate adjustment (variable speed coils only)
+            state.dataAirLoop->AirLoopFlow(airLoopNum).ReqSupplyFrac *=
+                this->m_CoolMassFlowRate[this->m_NumOfSpeedCooling] / state.dataAirLoop->AirLoopFlow(airLoopNum).DesSupply;
+        }
+        // determine air temperature difference across fan based on new mixed air flow rate
+        int mixedAirNode = outdoorAirController.MixNode;
+        Real64 fanDTAtMixedAirFlowRate = this->getFanDeltaTemp(
+            state, firstHVACIteration, mixedAirFlowRate, mixedAirFlowRate / this->m_CoolMassFlowRate[this->m_NumOfSpeedCooling]);
+        // determine new mixed air setpoint
+        Real64 newMixedAirSP = zoneTemp - std::abs(zoneLoad) / (cpAir * mixedAirFlowRate);
+        state.dataLoopNodes->Node(mixedAirNode).TempSetPoint = newMixedAirSP - fanDTAtMixedAirFlowRate;
+    }
+
+    void
+    UnitarySys::manageEconomizerStagingOperation(EnergyPlusData &state, int const airLoopNum, bool const firstHVACIteration, Real64 const zoneLoad)
+    {
+        if (airLoopNum > 0) {
+            if (state.dataAirLoop->AirLoopControlInfo(airLoopNum).EconoActive == true && state.dataGlobal->WarmupFlag == false &&
+                state.dataUnitarySystems->CoolingLoad) {
+                this->setEconomizerStagingOperationSpeed(state, firstHVACIteration, zoneLoad);
+
+                // adjustments are only needed when economizer speed is greater than the lowest speed
+                if (this->m_EconoSpeedNum > 0) {
+                    this->calcMixedTempAirSPforEconomizerStagingOperation(state, airLoopNum, firstHVACIteration, zoneLoad);
+
+                    // recalculate the outdoor air fraction to meet the new mixed air setpoint at the new mixed air flow rate
+                    MixedAir::ManageOutsideAirSystem(
+                        state, state.dataAirLoop->OutsideAirSys(this->OASysIndex).Name, firstHVACIteration, airLoopNum, this->OASysIndex);
```

Comment:
Oh, I see, the OA system gets called from the UnitarySystem. I previously thought that only the air loop control properties were set in UnitarySystem. So that means the mixed air set point IS getting used, and that calculation of that SP is necessary. This is pointing to a new report for "Unitary System Mixed Air Setpoint Temperature".

Reply:
I was just going through your comments one more time, so just to confirm, there's is nothing to add here, is that correct? The user should be able to get the setpoint set by the unitary system by requesting the system node setpoint temperature output variable.

Reply:
I think viewing the SP on the node will be sufficient for now.